lib/fencing/st_client.c

```diff
@@ -1757,17 +1762,21 @@ stonith_api_validate(stonith_t *st, int call_options, const char *rsc_id,
         *error_output = NULL;
     }
 
+    if (timeout_sec <= 0) {
+        timeout_sec = CRMD_METADATA_CALL_TIMEOUT; // Questionable
```

Comment:
Validate calls metadata only if host_arg is NULL, so it can check for port or plug support. Between that and being RHCS-specific I'd rather not cater to it much.

There's no obvious default here (even CRMD_METADATA_CALL_TIMEOUT is arbitrary for metadata actions). The scheduler's default timeout for actions could make sense, though this is only used by stonith_admin and doesn't have any real relation to anything else. Ultimately it's not terribly important, both metadata and validate are typically very quick, and if they're not something is probably wrong and they'll time out no matter what.